README.md

```diff
@@ -8,33 +8,103 @@ This crate provides a Rust implementation of the Schnorr signature scheme for th
 ## About
 The Schnorr signature scheme, named after its creator Claus Schnorr, is a digital signature scheme renowned for its simplicity. The scheme provides a simple method of creating short signatures. 
 
-The implementation has been created using the
-Poseidon hash function, the paper for which can
-be found [here](https://eprint.iacr.org/2019/458.pdf).
+The implementation has been created using the [`jubjub`](https://github.com/dusk-network/jubjub) elliptic curve and the [`Poseidon`](https://github.com/dusk-network/Poseidon252) hash function, the paper for which can be found [here](https://eprint.iacr.org/2019/458.pdf).
 
-For a reference to the algorithm, please see the 
-[docs](https://app.gitbook.com/@dusk-network/s/specs/proposals/rfc/rfc-17).
+The signature scheme is implemented within the [Phoenix](https://github.com/dusk-network/phoenix-core/blob/master/docs/protocol-description.pdf) transaction model and is based on the Schnorr Sigma protocol, compiled alongside the Fiat–Shamir transformation, to serve as a non-interactive signature scheme. Specifically, the Phoenix protocol employs a variant that utilizes double Schnorr signatures, verifiable with double public keys, enabling the delegation of computational processes within the protocol's later stages.
 
 ## Library Structure
-The library is partitioned into two components:
+The library is partitioned into the following components:
 
-- **Keys**: Module containing the secret key structure for signing messages, and the public key & keypair structures used in verification.
-- **Signatures**: Module containing functions to verify the validity of Schnorr signatures.
-- **Gadgets**: Contains the Plonk gadgets for signature verification.
+- **Keys**: Module containing the secret key structure for signing messages, as well as the public key and the double public key structures used in verification.
+- **Signatures**: Module containing the standard and double signature structs as well as functions to verify the validity of Schnorr signatures and double Schnorr signatures.
+- **Gadgets**: Contains the Plonk gadgets for in-circuit verification of Schnorr signatures and double Schnorr Signatures.
 
-## Schnorr Sigma Protocol
+## Signature Scheme Description
 
-The signature scheme implemented within the Phoenix protocol is based on the Schnorr Sigma protocol, compiled alongside the Fiat–Shamir transformation, to serve as a non-interactive signature scheme. Specifically, the Phoenix protocol employs a variant that utilizes double keys, enabling the delegation of computational processes within the protocol's later stages.
+### Notation
 
-### Signature Scheme Description
+In the following:
+- Multiplication of a point $P$ by a scalar $s$ stands for adding $P$ $s$-times to itself.
+- $\mathbb{Z}_q$ the set of congruence classes modulo $q$
+- for prime $q$: $\mathbb{Z}_q^× =  \mathbb{Z}_q \setminus \overline{0}_q$, all congruence classes modulo $q$ without the congruence class of $0$ modulo $q$.
```

Comment:
~~for prime [...] -> **F**or **a** prime $q$: $\mathbb{Z}_q^× =  \mathbb{Z}_q \setminus \overline{0}_q$ **denotes the set of** all congruence classes [...].~~

Following my previous comment:
- For a prime $q$: $\mathbb{F}_q^× =  \mathbb{F}_q \setminus \{ 0 \}$ contains of all nonzero elements of $\mathbb{F}_q$.